[cmake] Either skip USE_* or adjust the defaults to the ones of DUNE's cmake buildsystem

This has cost me and some others (see OPM/opm-grid#561) quite some time for newer DUNE versions.

We have USE_MPI, USE_SUPERLU, USE_QUADMATH, that default to off. Being off will result in OPM not searching for these packages. As result none of the important targets are available in OPM. The problem is that DUNE will as default always search for them and use them. That means libraries might link to them and will use imported targets in the package configuration files. This will result in
cmake errors if the user has not manually chosen DUNE's default.

To fix this we should at least use DUNE's defaults or just get rid off these settings. This is not a problem, because:
- USE_MPI is off for historic reasons when some in the project thought that parallel computing was not useful. Luckily this changed. Let's change the default, too.
- SuperLU is also off for historic reasons to prefer UMFPack which gave better performance. This is not needed anymore as DUNE will prefer UMFPack if available in AMG
- QuadMath  is not used by us and we just stayed consistent and used off as default.

Note that the latter two are only an issue if someone has dune-fem on their system.

Just found out that with the default USE_MPI=OFF we already fail for opm-common:
```
-- This build defaults to installing in /home/mblatt/opt/opm
CMake Error at opm-common/cmake/Modules/OpmSatellites.cmake:63 (add_executable):
  Target "test_cartgrid" links to target "MPI::MPI_C" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  opm-common/cmake/Modules/OpmLibMain.cmake:248 (opm_compile_satellites)
  CMakeLists.txt:119 (include)
```

--- a/cmake/Modules/Finddune-istl.cmake
+++ b/cmake/Modules/Finddune-istl.cmake
@@ -19,7 +19,7 @@ find_opm_package (
   "dune-common REQUIRED;
   ParMETIS;
   SuperLU;
-  SuiteSparse COMPONENTS umfpack
+  SuiteSparse COMPONENTS umfpack REQUIRED
   "
   # header to search for
   "dune/istl/bcrsmatrix.hh"

--- a/cmake/Modules/OpmInit.cmake
+++ b/cmake/Modules/OpmInit.cmake
@@ -130,7 +130,7 @@ endif ()
 
 # parallel computing must be explicitly enabled
 # This needs to be in OpmInit as prereqs is called before OpmLibMain is included.
-option (USE_MPI "Use Message Passing Interface for parallel computing" OFF)
+option (USE_MPI "Use Message Passing Interface for parallel computing" ON)
 if (NOT USE_MPI)
   set (CMAKE_DISABLE_FIND_PACKAGE_MPI TRUE)
 endif ()
@@ -144,7 +144,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 # quadmath must be explicitly enabled
 # This needs to be in OpmInit as prereqs is called before OpmLibMain is included.
-option (USE_QUADMATH "Use high precision floating point library (slow)" OFF)
+option (USE_QUADMATH "Use high precision floating point library (slow)" ON)
 if (NOT USE_QUADMATH)
   set (CMAKE_DISABLE_FIND_PACKAGE_QuadMath TRUE)
 endif ()

--- a/cmake/Modules/OpmLibMain.cmake
+++ b/cmake/Modules/OpmLibMain.cmake
@@ -64,7 +64,7 @@ include (UseThreads)
 find_threads (${project})
 
 # SuperLU is optional
-option (USE_SUPERLU "Use SuperLU direct solvers" OFF)
+option (USE_SUPERLU "Use SuperLU direct solvers for AMG (if umfpack is not found)" ON)
 
 # PETSc is optional
 option (USE_PETSC "Use PETSc iterative solvers" OFF)

